lib/src/framework.dart

```diff
@@ -225,6 +226,12 @@ abstract class HookState<R, T extends Hook<R>> with Diagnosticable {
   @protected
   void initHook() {}
 
+  /// Register a callback for when [HookWidget.build] is finished
+  void setDidBuildListener(VoidCallback callback) {
+    _didBuildCallback = callback;
+    _element?.addHookForDidBuild(this);
```

Comment:
We'd need to override dispose to unsubscribe the hook

Reply:
True, totally forget that we would need to remove them at some point, I'll add a `removeHookForDidBuild`

Reply:
To we want to call `removeHookForDidBuild ` automatically on the default `dispose` or do we let dev call it manually by overriding `dispose` ? For example if `_didBuildCallback` become nullable we can call `removeHookForDidBuild ` automatically on `dispose` if it's not null. It will prevent dev from forgetting to remove their hook in dispose

Reply:
I wanted to do something like this:
```
/// Equivalent of [State.dispose] for [HookState]
  @protected
  @mustCallSuper
  void dispose() {
    if (_didBuildCallback != null) {
      _didBuildCallback = null;
      _element?.removeHookForDidBuild(this);
    }
  }
```
But by doing so it triggers a lot of warning for existing Hook that doesn't call super. I prefer doing it automatically but if you think it's too much I'll just put on the doc that they should call remove on the on dispose.

Reply:
I've not done that because if will be a change for all existing hooks, instead I have said on the doc that remove should be called if dispose is overridden